### PR TITLE
Reduce CI load by cancelling previous runs with every new commit in PR

### PR DESCRIPTION
**Pull request by stloyd**
<h2>Change Log</h2>
<div id="change-log">
  <h4>Added</h4>
  <ul id="added">
    
  </ul> 
  <h4>Fixed</h4>  
  <ul id="fixed">
    
  </ul>
  <h4>Changed</h4>
  <ul id="changed">
    <li>Reduce CI load by cancelling previous runs with every new commit in PR</li>
  </ul>  
  <h4>Removed</h4>
  <ul id="removed">
    
  </ul>
  <h4>Deprecated</h4>
  <ul id="deprecated">
    
  </ul>  
  <h4>Security</h4> 
  <ul id="security">
    
  </ul>     
</div>
<hr/>

<h2>Description</h2>

Based on: https://stackoverflow.com/questions/66335225/how-to-cancel-previous-runs-in-the-pr-when-you-push-new-commitsupdate-the-curre/72408109#72408109

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -2,10 +2,20 @@ name: Documentation Linter
 
 on:
   pull_request:
+    paths:
```
Comment (stloyd) [inline .github/workflows/docs.yml:5]: This change requires removing that workflow from required as it's not being run right now.

**Comment by norberttech** [comment]
hmm I'm getting job cancellation notifications, isn't that conflicting with post merge jobs also?

**Comment by stloyd** [comment]
Maybe just turn off those cause they are not useful as cancellation should happen on manual trigger or particular failure of infrastructure.

**Comment by norberttech** [comment]
oh I know what's happening, whenver we merge anything, automation is also pushing changes to the CHANGELOG.md file, which triggers all tests all over again. Could you take a look into this and try to figure out if there is any way to prevent triggering the whole testsuite after the update of CHANGELOG.md?

**Comment by stloyd** [comment]
@norberttech Should be fixed with: https://github.com/flow-php/flow/pull/623

**Pull request merged**